File: C++/Level1/Deck.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    int card1Index = 0, card2Index = 0;

    for (string& g : goal)
    {
        if (g == cards1[card1Index]) 
        {
            ++card1Index;
        }
        else if (g == cards2[card2Index]) 
        {
            ++card2Index;
        }
        else
        {
            return "No";
        }
    }

    return "Yes";
}

File: Java/Level1/Deck.java
/*
 * https://school.programmers.co.kr/learn/courses/30/lessons/159994?language=java
 */

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int index1 = 0, index2 = 0;

        for(int i = 0; i < goal.length; ++i) {
            if(index1 < cards1.length && goal[i].compareTo(cards1[index1]) == 0) {
                ++index1;
            }
            else if(index2 < cards2.length && goal[i].compareTo(cards2[index2]) == 0) {
                ++index2;
            }
            else {
                return "No";
            }
        }

        return "Yes";
    }
}